### PR TITLE
Bug Fixes for OBO credentials

### PR DESCRIPTION
**Pull request by vikasagrawal-ms**
Bug fixes
Added user_identity for OBO credentials in download component.
Updated get_ml_client method to first fetch MSI and then default to OBO credential. This is done to handle any user provided MSI

### CODE REVIEW

File: assets/common/components/register_model/spec.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 $schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
 name: register_model
-version: 0.0.8
+version: 0.0.8.reg2
```
Comment (novaturient95) [inline assets/common/components/register_model/spec.yaml:3]: lot of temp changes. Could you please remove them?

File: assets/training/model_management/components/import_model/spec.yaml [hunk 2]
```diff
@@ -175,7 +175,7 @@ outputs:
 
 jobs:
   validation_trigger_import:
-    component: azureml:validation_trigger_import:0.0.2
+    component: azureml://registries/azureml/components/validation_trigger_import/versions/0.0.2
```
Comment (novaturient95) [inline assets/training/model_management/components/import_model/spec.yaml:178]: remove

File: assets/training/model_management/components/import_model/spec.yaml [hunk 3]
```diff
@@ -208,10 +208,12 @@ jobs:
         type: uri_file
 
   download_model:
-    component: azureml:download_model:0.0.14
+    component: azureml://registries/azureml-preview-test1/components/download_model/versions/0.0.14.reg1
```
Comment (novaturient95) [inline assets/training/model_management/components/import_model/spec.yaml:211]: remove

File: assets/training/model_management/components/import_model/spec.yaml [hunk 4]
```diff
@@ -261,7 +263,7 @@ jobs:
       mlflow_model_folder: ${{parent.outputs.mlflow_model_folder}}
 
   register_model:
-    component: azureml:register_model:0.0.8
+    component: azureml://registries/azureml-preview-test1/components/register_model/versions/0.0.8.reg2
```
Comment (novaturient95) [inline assets/training/model_management/components/import_model/spec.yaml:266]: revert

**Review by novaturient95 — DISMISSED** [review]

**Comment by novaturient95** [comment]
Could you update description with successful run?

File: assets/common/components/register_model/spec.yaml (hunk 1 above)
Comment (vikasagrawal-ms) [reply assets/common/components/register_model/spec.yaml]: done

**Review by skasturi — APPROVED** [review]

**Pull request merged**